confignode/src/main/java/org/apache/iotdb/confignode/manager/partition/PartitionManager.java

```diff
@@ -754,69 +760,181 @@ public void maintainRegionReplicas() {
-                            break;
-
-                          case DataRegion:
-                          default:
-                            // Create DataRegion
-                            status =
-                                SyncDataNodeClientPool.getInstance()
-                                    .sendSyncRequestToDataNodeWithRetry(
-                                        createEntry.getTargetDataNode().getInternalEndPoint(),
-                                        new TCreateDataRegionReq(
-                                                createEntry.getRegionReplicaSet(),
-                                                createEntry.getStorageGroup())
-                                            .setTtl(createEntry.getTTL()),
-                                        DataNodeRequestType.CREATE_DATA_REGION);
-                        }
-                        break;
+                if (regionMaintainTaskList.isEmpty()) {
+                  return;
+                }
 
-                      case DELETE:
-                      default:
-                        // Delete Region
-                        RegionDeleteTask deleteEntry = (RegionDeleteTask) entry;
-                        LOGGER.info(
-                            "Start to delete Region: {} on DataNode: {}",
-                            deleteEntry.getRegionId(),
-                            deleteEntry.getTargetDataNode());
-                        status =
-                            SyncDataNodeClientPool.getInstance()
-                                .sendSyncRequestToDataNodeWithRetry(
-                                    deleteEntry.getTargetDataNode().getInternalEndPoint(),
-                                    deleteEntry.getRegionId(),
-                                    DataNodeRequestType.DELETE_REGION);
+                // group tasks by region id
+                Map<TConsensusGroupId, Queue<RegionMaintainTask>> regionMaintainTaskMap =
+                    new HashMap<>();
+                for (RegionMaintainTask regionMaintainTask : regionMaintainTaskList) {
+                  regionMaintainTaskMap
+                      .computeIfAbsent(regionMaintainTask.getRegionId(), k -> new LinkedList<>())
+                      .add(regionMaintainTask);
+                }
+
+                while (!regionMaintainTaskMap.isEmpty()) {
+
+                  // select same type task from each region group
+                  List<RegionMaintainTask> selectedRegionMaintainTask = new ArrayList<>();
+                  RegionMaintainType currentType = null;
+                  for (Map.Entry<TConsensusGroupId, Queue<RegionMaintainTask>> entry :
+                      regionMaintainTaskMap.entrySet()) {
+                    RegionMaintainTask regionMaintainTask = entry.getValue().peek();
+                    if (regionMaintainTask != null) {
+                      if (currentType == null
+                          || regionMaintainTask.getType().equals(RegionMaintainType.DELETE)
```

Comment:
Why type DELETE can always pass if? Therefore the delete tasks will always be added to selectedRegionMaintainTask?

Reply:
Fixed.

Reply:
Awesome! the division logic is correct~ By the way, could you please add more annotations for these division codes? Because they might be difficult for others to understand~